src/compiler.rs

```diff
@@ -135,6 +135,30 @@ pub fn compile(
         }
     }
 
+    {
+        // Add the builtin_costs global.
+        // We always add it because symbol look up otherwise can panic.
+        let region = Region::new();
+        let location = Location::unknown(context);
+        let block = region.append_block(Block::new(&[]));
+        let value = block.append_op_result(
+            ods::llvm::mlir_zero(context, llvm::r#type::pointer(context, 0), location).into(),
+        )?;
+        block.append_operation(melior::dialect::llvm::r#return(Some(value), location));
+
+        module.body().append_operation(
+            ods::llvm::mlir_global(
+                context,
+                region,
+                TypeAttribute::new(llvm::r#type::pointer(context, 0)),
+                StringAttribute::new(context, "builtin_costs"),
+                Attribute::parse(context, "#llvm.linkage<external>").unwrap(),
+                location,
+            )
+            .into(),
+        );
+    }
```

Comment:
Does it have to be a global? It won't cause an issue with neither the AOT (shared libraries only) or JIT, but it could be problematic when compiling a standalone object from multiple cairo programs due to symbol redefinitions.

Reply:
Yes it has to be a global, the builtin costs is not passed anywhere as an input.

> but it could be problematic when compiling a standalone object from multiple cairo programs due to symbol redefinitions.

Is this something we do? I think its out of scope anyway, projects / contracts always end up being 1 sierra file and thus 1 library

Reply:
The ones run with cairo-native-run yes, but we can also compile it into an object file and link it with other programs. I guess it's fine for now since that use case is so rare and not our priority. Also I don't see how we could do this without a global or a big refactor.